app/src/main/java/org/oppia/app/player/state/StateAdapter.kt

```diff
@@ -29,14 +30,21 @@ import org.oppia.app.player.state.itemviewmodel.StateButtonViewModel
 import org.oppia.app.player.state.itemviewmodel.TextInputInteractionViewModel
 import org.oppia.app.player.state.listener.InputInteractionTextListener
 import org.oppia.app.player.state.listener.InteractionListener
+import org.oppia.util.parser.HtmlParser
 
 const val VIEW_TYPE_CONTENT = 1
 const val VIEW_TYPE_INTERACTION_READ_ONLY = 2
 const val VIEW_TYPE_NUMERIC_INPUT_INTERACTION = 3
 const val VIEW_TYPE_TEXT_INPUT_INTERACTION = 4
 const val VIEW_TYPE_STATE_BUTTON = 5
 
-class StateAdapter(private val itemList: MutableList<Any>, private val interactionListener: InteractionListener) :
+class StateAdapter(
+  private val itemList: MutableList<Any>,
+  private val interactionListener: InteractionListener,
+  private val htmlParserFactory: HtmlParser.Factory,
+  private val entityType: String,
```

Comment:
Do we need to pass this in if we know that the entity ID is an exploration ID? I'm guessing we can infer at this point that the entity type will be exploration.

Reply:
To use it in HtmlParser, that is reusable class. I am specifying `entityType` to "exploration" because HtmlParser takes it as parameter to load images ,if its not from exploration and any other entity we can pass that string.

Reply:
Injected entityType.